### ISSUE TITLE
adler32 hash is too small ==> Many collisions

### ISSUE THREAD

**Issue by louismartin** [shaypal5/cachier#19]
Hi, 
I just updated to the cachier 1.3.3 with the adler32 hashing and I have many bugs related to hash collision.
For instance I have a bunch of 2950 paths that hash to only 1195 unique hash values which causes my method to return a lot of erroneous values (i.e. cached values corresponding to another path).

Here's a simple illustration:
```
from zlib import adler32
import pickle

hashes = [adler32(pickle.dumps(str(i))) & 0xffffffff for i in range(1000)]
print(len(hashes), len(set(hashes)))

Output: 1000 424
```

The computed hash seems to be too short (10 digits) and not hexadecimal.
There might be a better alternative with longer hexadecimal hashes.

**Comment by louismartin** [comment]
An idea of a test that could be added to catch this problem (fails with the current version):
```
from cachier import cachier

@cachier(pickle_reload=False)
def test(i):
    return i

for _ in range(2):
    for i in range(10000):
        assert test(str(i)) == str(i)
```

**Comment by shaypal5** [comment]
I rolled back all hash related changes, and released this under `v1.4.0`:
https://github.com/shaypal5/cachier/releases/tag/v1.4.0

I'm closing this. Please reopen if this does not resolve your issues.

**Comment by shaypal5** [comment]
Sorry, `v1.4.1` should do the trick.
